File: Assets/common/script/scenery/BlinkenFlareBehaviour.js
#pragma strict

var delay : float = 75.0f;
var blinking : boolean = true;
var offset : float = 0.0f;
private var lastTime : float;
var startDelay : float = 75.0f;

var dying : boolean = false;
var deadTime : float = 0;

var target : Transform;

private var theShip : Transform;
private var mat : Material;
private var state: boolean  =false;


function Start () {
	
	theShip = GameObject.Find("TheShip").transform;
	
	mat = GetComponentInChildren(Renderer).material;
//	Debug.Log(mat);
	if(blinking == false){
		mat.color.a = 0.0;
	}
}

function flickerAndDie(){
	dying = true;
	deadTime = Time.fixedTime;
}

function Update () {
	if(blinking){
		
			
		
		if(target == null){
			transform.LookAt(theShip);
		} else {
			transform.LookAt(target);
		}
	} 
}

function FixedUpdate(){
	if(blinking){
		delay --;
		if(delay < 0){
			delay = startDelay;
			state = ! state;
		}
		
			
		if(state){	
			mat.color.a = 1;
		} else {
			mat.color.a = 0.0f;
		}
		
	}
	if(dying ){
		if( deadTime + 1.5f < Time.fixedTime){
			blinking = false;
			dying = false;
			mat.color.a = 0;
		} else {
			mat.color.a = Random.value;
		}
	}
}

File: Assets/common/script/scenery/BlinkenFlareBehaviour.cs
using UnityEngine;
using System;


public class BlinkenFlareBehaviour:MonoBehaviour{
	
	public float delay = 75.0f;
	public bool blinking = true;
	public float offset = 0.0f;
	float lastTime;
	public float startDelay = 75.0f;
	
	public bool dying = false;
	public float deadTime = 0.0f;
	
	public Transform target;
	
	Transform theShip;
	Material mat;
	bool state  =false;
	
	
	public void Start() {

		theShip = GameObject.Find("TheShip").transform;
		if (target == null) {
			target = theShip;
		}
		mat = GetComponentInChildren<Renderer>().material;
	//	Debug.Log(mat);
		if(blinking == false){
			var tmp_cs1 = mat.color;
            tmp_cs1.a = 0.0f;
            mat.color = tmp_cs1;
		}
	}
	
	public void flickerAndDie(){
		dying = true;
		deadTime = Time.fixedTime;
	}
	
	public void Update() {
		if(blinking){
			
				
			
			if(target == null){
				transform.LookAt(theShip);
			} else {
				transform.LookAt(target);
			}
		} 
	}
	
	public void FixedUpdate(){
		if(blinking){
			delay --;
			if(delay < 0){
				delay = startDelay;
				state = ! state;
			}
			
				
			if(state){	
				var tmp_cs2 = mat.color;
                tmp_cs2.a = 1.0f;
                mat.color = tmp_cs2;
			} else {
				var tmp_cs3 = mat.color;
                tmp_cs3.a = 0.0f;
                mat.color = tmp_cs3;
			}
			
		}
		if(dying ){
			if( deadTime + 1.5f < Time.fixedTime){
				blinking = false;
				dying = false;
				var tmp_cs4 = mat.color;
                tmp_cs4.a = 0.0f;
                mat.color = tmp_cs4;
			} else {
				var tmp_cs5 = mat.color;
                tmp_cs5.a = UnityEngine.Random.value;
                mat.color = tmp_cs5;
			}
		}
	}
}